src/software/ai/play_selection_fsm.cpp

```diff
@@ -41,45 +41,70 @@ bool PlaySelectionFSM::gameStateSetupRestart(const Update& event)
 
 void PlaySelectionFSM::setupSetPlay(const Update& event)
 {
-    current_play.reset();
     if (event.game_state.isOurBallPlacement())
     {
-        event.set_current_play(std::make_unique<BallPlacementPlay>(ai_config));
+        if (current_set_play != TbotsProto::PlayName::BallPlacementPlay)
+        {
+            current_set_play = TbotsProto::PlayName::BallPlacementPlay;
+            event.set_current_play(std::make_unique<BallPlacementPlay>(ai_config));
+        }
```

Comment:
nit: you could probably merge it with the `if` conditional

`if (event.game_state.isOurBallPlacement() && current_set_play != TbotsProto::PlayName::BallPlacementPlay)`

Reply:
If `event.game_state.isOurBallPlacement()` is true, we don't want to enter any of the other `else if` branches if `current_set_play == TbotsProto::PlayName::BallPlacementPlay)`